android/app/src/main/kotlin/io/lantern/model/SessionModel.kt

```diff
@@ -161,11 +161,23 @@ class SessionModel(
             }
 
 
-            "submitGooglePlayPayment" -> paymentsUtil.submitGooglePlayPayment(
-                call.argument("email")!!,
-                call.argument("planID")!!,
-                result,
-            )
+            "submitGooglePlayPayment" -> {
+                try {
+                    paymentsUtil.submitGooglePlayPayment(
+                        call.argument("email")!!,
+                        call.argument("planID")!!,
+                        result,
+                    )
+                }
+                catch (e: Exception) {
+                    result.error(
```

Comment:
Do you want to log the exception thrown here?

Reply:
We do, so if something goes wrong, it should send the error to Flutter and update the UI according. Before it was just loading forever

Reply:
I thought the error being shown is R.string.error_making_purchase and not the actual message from the exception

Reply:
@atavism Added log exception.